string state
int64 value